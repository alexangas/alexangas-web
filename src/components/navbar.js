import React from "react"
import { useStaticQuery, graphql, Link } from "gatsby"

export const PureNavBar = ({ data }) => {
  const { title } = data.site.siteMetadata

  return (
    <div className="navbar-wrapper">
      <nav
        className="container navbar is-transparent"
        role="navigation"
        aria-label="Main navigation"
      >
        <div className="navbar-menu is-active">
          <div className="navbar-start is-size-4">
            <Link to="/" className="navbar-item has-text-weight-semibold">
              {title}
            </Link>
          </div>

          <div className="navbar-end is-size-5">
            <Link to="/blog/" className="navbar-item">
              Blog
            </Link>
          </div>
        </div>
      </nav>
    </div>
  )
}

export const NavBar = (props) => {
  const data = useStaticQuery(graphql`
    query NavBarQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return <PureNavBar {...props} data={data} />
}

export default NavBar
