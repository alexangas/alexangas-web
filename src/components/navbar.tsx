import * as React from "react"
import { graphql, Link, useStaticQuery } from "gatsby"
import clsx from "clsx"

type NavBarProps = {
  location: Location
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  data?: any
}

export const PureNavBar = ({ location, data }: NavBarProps): JSX.Element => {
  const locationIsActive = (key: string): boolean => {
    const locationMap: Record<string, string> = {
      Blog: `/blog`,
      Contact: `/contact`,
    }

    return location.pathname.startsWith(locationMap[key])
  }

  return (
    <div className="navbar-wrapper">
      <nav
        className="container navbar is-transparent"
        role="navigation"
        aria-label="Main navigation"
      >
        <div className="navbar-menu is-active px-2 py-2">
          <div className="navbar-start is-size-4">
            <Link to="/" className="navbar-item has-text-weight-semibold">
              {data.site.siteMetadata.title}
            </Link>
          </div>

          <div className="navbar-end is-size-5">
            <Link
              to="/contact/"
              className={clsx(
                `navbar-item`,
                `is-tab`,
                locationIsActive(`Contact`) && `is-active`
              )}
            >
              Contact
            </Link>
            <Link
              to="/blog/"
              className={clsx(
                `navbar-item`,
                `is-tab`,
                locationIsActive(`Blog`) && `is-active`
              )}
            >
              Blog
            </Link>
          </div>
        </div>
      </nav>
    </div>
  )
}

export const NavBar = (props: NavBarProps): JSX.Element => {
  const data = useStaticQuery(graphql`
    query NavBarQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return <PureNavBar {...props} data={data} />
}

export default NavBar
