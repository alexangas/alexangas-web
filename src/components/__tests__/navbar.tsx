import React from "react"
import renderer from "react-test-renderer"

import { PureNavBar as NavBar } from "../navbar"

describe(`NavBar`, () => {
  it(`renders title`, () => {
    const location: Partial<Location> = {
      pathname: `nowhere`
    }

    const data = {
      site: {
        siteMetadata: {
          title: `Site Title`,
        },
      },
    }

    const testRenderer = renderer.create(
      <NavBar location={location as Location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders Contact`, () => {
    const location: Partial<Location> = {
      pathname: `/contact`
    }

    const data = {
      site: {
        siteMetadata: {
          title: `Contact`,
        },
      },
    }

    const testRenderer = renderer.create(
      <NavBar location={location as Location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`renders Blog`, () => {
    const location: Partial<Location> = {
      pathname: `/blog/post`
    }

    const data = {
      site: {
        siteMetadata: {
          title: `Blog Post`,
        },
      },
    }

    const testRenderer = renderer.create(
      <NavBar location={location as Location} data={data} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
