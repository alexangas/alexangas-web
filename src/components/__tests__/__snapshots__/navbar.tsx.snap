// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders title 1`] = `
<div
  className="navbar-wrapper"
>
  <nav
    aria-label="Main navigation"
    className="container navbar is-transparent"
    role="navigation"
  >
    <div
      className="navbar-menu is-active"
    >
      <div
        className="navbar-start is-size-4"
      >
        <a
          className="navbar-item has-text-weight-semibold"
          href="/"
        >
          Site Title
        </a>
      </div>
      <div
        className="navbar-end is-size-5"
      >
        <a
          className="navbar-item"
          href="/blog/"
        >
          Blog
        </a>
      </div>
    </div>
  </nav>
</div>
`;
